Sample Application/SampleApplication/Screens/CardList/CustomViews/CardDetailView.swift

```diff
@@ -0,0 +1,249 @@
+        case .active:
+            return (Constants.activateIcon, Constants.activateText, $isActivating)
+        case .suspended:
+            return (Constants.suspendIcon, Constants.suspendText, $isSuspending)
+        case .revoked:
+            return (Constants.revokeIcon, Constants.revokeText, $isRevoking)
+
+        default:
+            return (String(), String(), .constant(false))
+        }
+    }
+
+    @ViewBuilder
+    private func buttonChangingState(to newState: CardState) -> some View {
+        let config = buttonConfig(for: newState)
+
+        loadingButton(isLoading: config.loadingState) {
+            delegate?.changeCardState(to: newState, completion: {
+                config.loadingState.wrappedValue = false
+
+                withAnimation {
+                    canAddToWallet = newState == .active
+                }
+            })
+        } label: {
+            HStack {
+                Image(systemName: config.image)
+                Text(config.title)
+            }
+            .frame(maxWidth: .infinity, maxHeight: .infinity)
+        }
+        .actionButtonStyle(isDestructive: newState == .revoked)
+    }
+
+    @ViewBuilder
+    private func loadingButton(isLoading: Binding<Bool>,
+                               action: @escaping () -> Void,
+                               @ViewBuilder label: @escaping () -> some View) -> some View {
+        Button(action: {
+            isLoading.wrappedValue.toggle()
+            action()
+        }) {
+            if isLoading.wrappedValue {
+                ProgressView()
+                    .progressViewStyle(CircularProgressViewStyle())
+            } else {
+                label()
+            }
+        }
+        .disabled(isLoadingInProgress)
+    }
+
+    enum Constants {
+        static let panText = "Card Number"
+        static let panPlaceholder = "•••• •••• •••• ••••"
+        static let securityCodeText = "Security Code"
+        static let securityCodePlaceholder = "•••"
+        static let expiryDateText = "Expiry Date"
+        static let showPinText = "Show PIN"
+        static let activateText = "Activate"
+        static let suspendText = "Suspend"
+        static let revokeText = "Revoke"
+
+        static let pinIcon = "lock"
+        static let activateIcon = "bolt"
+        static let suspendIcon = "pause"
+        static let revokeIcon = "power"
+        static let showDetailsIcon = "eye"
+        static let hideDetailsIcon = "eye.slash"
+
+        static let buttonHeight: CGFloat = 52
+        static let infoRowHeight: CGFloat = 20
+    }
+}
+
+fileprivate extension View {
+    func actionButtonStyle(isDestructive: Bool = false) -> some View {
+        modifier(CardDetailView.ActionButton(isDestructive: isDestructive))
+    }
+}
+
+fileprivate extension Text {
+    func titleStyle() -> some View {
+        modifier(CardDetailView.Title())
+    }
+
+    func blurred() -> some View {
+        modifier(CardDetailView.BlurredText())
+    }
```

Comment:
Generally these layers that make handling modifiers more generic should be placed in separate files. I would move them to separate files under `/Models/ViewExtensions`?

Alternatively, just move these modifiers to the `CardDetailView+ViewModifiers`?

Reply:
I would avoid doing this since they are extensions for common types but at the same time using definitions from `CardDetailView`. Otherwise it would confuse the consumers of, let's say, `Text` as blurred means a generic blurring but it's actually specific to `CardDetailView`.